server/verify.go

```diff
@@ -65,6 +65,22 @@ func VerifyAttestation(attestation *pb.Attestation, opts VerifyOpts) (*pb.Machin
 	if err != nil {
 		return nil, fmt.Errorf("failed to get AK public key: %w", err)
 	}
+
+	// Add intermediate certs in the attestation if they exist.
+	if len(attestation.IntermediateCerts) != 0 {
+		if opts.IntermediateCerts == nil {
+			opts.IntermediateCerts = x509.NewCertPool()
+		}
+
+		for _, certBytes := range attestation.IntermediateCerts {
+			cert, err := x509.ParseCertificate(certBytes)
+			if err != nil {
+				return nil, fmt.Errorf("filed to parse intermediate certificate in attestation: %w", err)
```

Comment:
I was unsure if this should raise an error or proceed without adding a certificate (and let any validation issues be caught in `checkAKTrusted`).

Reply:
I'm inclined to fail since the client-side already does this check. The proto should not have a malformed intermediate cert. This mirrors the behavior on a malformed AK in `checkAKTrusted`
Note that `ParseCertificate` returns nil on a parsing error. https://cs.opensource.google/go/go/+/refs/tags/go1.17.8:src/crypto/x509/parser.go;l=989
If we then feed nil into `AddCert`, it will panic. So we need to handle the nil case properly either way (e.g., if we decide to not error out for whatever reason, we can skip)

also typo in the error (failed)